### PR TITLE
fix: check jwt token  and add role in token

### PR DESCRIPTION
**Pull request by Lucas-Malatchoumy**
### User

- Add jwt verification in update delete and getters methods
- Check if the current user update his own profile
- Check if the current user delete his own profile
- Check if user has SUPER_ADMIN or ADMIN role to get all users

### CODE REVIEW

File: user/views.py [hunk 1]
```diff
@@ -58,8 +59,9 @@ def login(request):
         })
 
     data = user_normalizer(user)
+    # user_role =
```
Comment (Maengdok) [inline user/views.py]: Petit oubli à retirer

File: user/views.py [hunk 2]
```diff
@@ -58,8 +59,9 @@ def login(request):
         })
 
     data = user_normalizer(user)
+    # user_role = 
     key = TOKEN_KEY
-    token = jwt.encode({"id": user.id}, key, algorithm="HS256")
+    token = jwt.encode({"id": user.id, "role": data["role"]}, key, algorithm="HS256")
```
Comment (Maengdok) [inline user/views.py]: N'oublies pas de variabiliser ton object pour que la lecture des paramètres de la fonction soit plus facile.

File: user/views.py [hunk 3]
```diff
@@ -137,7 +157,7 @@ def register(request):
     data = user_normalizer(User.objects.latest('id'))
     # Change key
     key = TOKEN_KEY
-    token = jwt.encode({"id": new_user.id}, key, algorithm="HS256")
+    token = jwt.encode({"id": new_user.id, "role": new_user.role.label}, key, algorithm="HS256")
```
Comment (Maengdok) [inline user/views.py]: Pareil içi, a variabiliser

**Review by Maengdok — CHANGES_REQUESTED** [review]

File: user/views.py (hunk 3 above)
Comment (Lucas-Malatchoumy) [reply user/views.py]: [9595ef3](https://github.com/Axel-Pion-MDS/M1_Project_Timer_Back/pull/11/commits/9595ef3ca184808fd4e03ffebc9ab22f262ea4a1)

File: user/views.py (hunk 2 above)
Comment (Lucas-Malatchoumy) [reply user/views.py]: [9595ef3](https://github.com/Axel-Pion-MDS/M1_Project_Timer_Back/pull/11/commits/9595ef3ca184808fd4e03ffebc9ab22f262ea4a1)

File: user/views.py (hunk 1 above)
Comment (Lucas-Malatchoumy) [reply user/views.py]: [9595ef3](https://github.com/Axel-Pion-MDS/M1_Project_Timer_Back/pull/11/commits/9595ef3ca184808fd4e03ffebc9ab22f262ea4a1)

**Review by Maengdok — APPROVED** [review]

**Review by elton-dy — APPROVED** [review]

**Pull request merged**